frontend/src/vault/keysign/shared/utils/extractAndFormatFees.ts

```diff
@@ -0,0 +1,81 @@
+import { fromChainAmount } from '../../../../chain/utils/fromChainAmount';
+import { KeysignPayload } from '../../../../gen/vultisig/keysign/v1/keysign_message_pb';
+import { formatAmount } from '../../../../lib/utils/formatAmount';
+
+export const extractAndFormatFees = ({
+  blockchainSpecific,
+  currency,
+  decimals,
+  locale = 'en-US',
+}: {
+  blockchainSpecific: KeysignPayload['blockchainSpecific'];
+  currency: string;
+  decimals?: number;
+  locale?: string;
+}) => {
+  if (!blockchainSpecific || !decimals) {
+    return {
+      networkFeesFormatted: null,
+      totalFeesFormatted: null,
+    };
+  }
+
+  let networkFee = null;
+  let totalFee = null;
+
+  switch (blockchainSpecific.case) {
```

Comment:
Better use `match` function instead of switch-case, and I believe we already have this logic in `chain/tx/fee`